iceoryx_posh/include/iceoryx_posh/internal/popo/ports/subscriber_port_user.hpp

```diff
@@ -0,0 +1,101 @@
+/// The SubscriberPortUser uses the functionality of a ChunkReceiver for receiving shared memory chunks. Additionally it
+/// provides the subscribe / unsubscribe API which controls whether the subscriber ports shall try to subscribe to
+/// matching publisher ports
+class SubscriberPortUser
+{
+  public:
+    using MemberType_t = SubscriberPortData;
+
+    explicit SubscriberPortUser(cxx::not_null<MemberType_t* const> subscriberPortDataPtr) noexcept;
+
+    SubscriberPortUser(const SubscriberPortUser& other) = delete;
+    SubscriberPortUser& operator=(const SubscriberPortUser&) = delete;
+    SubscriberPortUser(SubscriberPortUser&& rhs) = default;
+    SubscriberPortUser& operator=(SubscriberPortUser&& rhs) = default;
+    ~SubscriberPortUser() = default;
+
+    /// @brief try to subscribe to all matching publishers
+    /// @param[in] queueCapacity, capacity of the queue where chunks are stored before they are passed to the user with
+    /// getChunk. Caution: Depending on the underlying queue there can be a different overflow behavior
+    void subscribe(const uint32_t queueCapacity = MAX_RECEIVER_QUEUE_CAPACITY) noexcept;
+
+    /// @brief unsubscribe from publishers, if there are any to which we are currently subscribed
+    void unsubscribe() noexcept;
+
+    /// @brief get the current subscription state. Caution: There can be delays between calling subscribe and a change
+    /// in the subscription state. The subscription state can also change without user interaction if publishers come
+    /// and go
+    /// @return SubscribeState
+    SubscribeState getSubscriptionState() const noexcept;
+
+    /// @brief Tries to get the next chunk from the queue. If there is a new one, the ChunkHeader of the oldest chunk in
+    /// the queue is returned (FiFo queue)
+    /// @return optional that has a new chunk header or no value if there are no new chunks in the underlying queue,
+    /// ChunkReceiveError on error
+    cxx::expected<cxx::optional<const mepoo::ChunkHeader*>, ChunkReceiveError> getChunk() noexcept;
+
+    /// @brief Release a chunk that was obtained with getChunk
+    /// @param[in] chunkHeader, pointer to the ChunkHeader to release
+    void releaseChunk(const mepoo::ChunkHeader* chunkHeader) noexcept;
+
+    /// @brief Release all the chunks that are currently queued up.
+    void releaseQueuedChunks() noexcept;
+
+    /// @brief check if there are chunks in the queue
+    /// @return if there are chunks in the queue return true, otherwise false
+    bool hasNewChunks() noexcept;
+
+    /// @brief check if there was a queue overflow since the last call of hasLostChunks
+    /// @return true if the underlying queue overflowed since last call of this method, otherwise false
+    bool hasLostChunks() noexcept;
+
+    /// @todo we first need the new condition variable
+    void attachConditionVariable() noexcept;
+    void detachConditionVaribale() noexcept;
+    bool isConditionVariableAttached() noexcept;
```

Comment:
Please change return value to `bool` as it will later derive from `Condition` which provides these three methods, see https://github.com/mossmaurice/iceoryx/blob/iox-%2325-semaphore-refactoring/iceoryx_posh/include/iceoryx_posh/internal/popo/waitset/condition.hpp#L40

Reply:
I think we can change this when merging with your branch. In my branch there is also no ConditionVariableData yet, so I cannot finalize the signature anyway